docs/resources/repository_conf_auth.md

```diff
@@ -37,7 +37,10 @@ resource "cyral_repository_conf_auth" "some_resource_name" {
   - `ACCESS_TOKEN`
   - `AWS_IAM`
 - `client_tls` (String) Is the repo Client using TLS? Default is "disable".
-- `identity_provider` (String) The ID (Alias) of the identity provider integration.
+- `identity_provider` (String) The semantics of this field are as follows:
+  If not supplied, then end-user authentication is disabled.
+  If end-user authentication with Cyral Access Token is desired, then this must be any non-empty string (the exact value is irrelevant).
+  If end-user authentication with AWS IAM is desired, then this must be the ID of an AWS IAM integration, and the <auth_type> must be set to AWS_IAM.
```

Comment:
@ricardorey10 this is how it should *look like*, so you will have to adjust your `go` file to be able to generate such output:

```suggestion
- `identity_provider` (String) The semantics of this field changed in control planes `v4.14` and later. See how it should be configured depending on your control plane version:
  - `v4.13` and below:
     - Provide the ID (Alias) of the identity provider integration to allow user authentication using an IdP.
  - `v4.14` and later:
     - If not supplied, then end-user authentication is disabled.
     - If end-user authentication with Cyral Access Token is desired, then set to `ACCESS_TOKEN` or any other non-empty string.
     - If end-user authentication with AWS IAM is desired, then this must be the ID of an AWS IAM integration, and the `auth_type` attribute must be set to `AWS_IAM`.
```

This will generate this output, which is much easier to read: 
![image](https://github.com/cyralinc/terraform-provider-cyral/assets/796900/2b55e31e-5bd9-4b89-9f8d-40009316a1f6)

Use the [Terraform docs preview](https://registry.terraform.io/tools/doc-preview) tool to see how your MD files will look like in the final docs shown in the Terraform registry.

Reply:
Thanks for the feedback! We're releasing this in 4.13, so I'll update the versions a bit, but otherwise I'll copy the text pretty much verbatim.

I'll make sure that the output looks correct before merging!